great_expectations/datasource/fluent/config.py

```diff
@@ -139,3 +177,239 @@ def parse_yaml(
+        )
+        yaml.dump(intermediate_json_dict, stream=stream_or_path, **yaml_kwargs)
+
+        if isinstance(stream_or_path, pathlib.Path):
+            return stream_or_path
+
+        return stream_or_path.getvalue()
+
+    @public_api
+    def json(
+        self,
+        *,
+        include: AbstractSetIntStr | MappingIntStrAny | None = None,
+        exclude: AbstractSetIntStr | MappingIntStrAny | None = None,
+        by_alias: bool = False,
+        # deprecated - use exclude_unset instead
+        skip_defaults: bool | None = None,
+        # Default to True to prevent serializing long configs full of unset default values
+        exclude_unset: bool = True,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        encoder: Callable[[Any], Any] | None = None,
+        models_as_dict: bool = True,
+        **dumps_kwargs: Any,
+    ) -> str:
+        """
+        Generate a JSON representation of the model, `include` and `exclude` arguments
+        as per `dict()`.
+        `encoder` is an optional function to supply as `default` to json.dumps(), other
+        arguments as per `json.dumps()`.
+        Deviates from pydantic `exclude_unset` `True` by default instead of `False` by
+        default.
+        """
+        intermediate_json_dict = self._json_dict(
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+            encoder=encoder,
+            models_as_dict=models_as_dict,
+        )
+        intermediate_json_dict = self._exclude_name_fields_from_fluent_datasources(
+            config=intermediate_json_dict
+        )
+        return self.__config__.json_dumps(
+            intermediate_json_dict, default=encoder, **dumps_kwargs
+        )
+
+    def _json_dict(
+        self,
+        *,
+        include: Union[AbstractSetIntStr, MappingIntStrAny, None] = None,
+        exclude: Union[AbstractSetIntStr, MappingIntStrAny, None] = None,
+        by_alias: bool = False,
+        exclude_unset: bool = True,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        encoder: Union[Callable[[Any], Any], None] = None,
+        models_as_dict: bool = True,
+        **dumps_kwargs,
+    ) -> dict:
+        """
+        JSON compatible dictionary. All complex types removed.
+        Prefer `.dict()` or `.json()`
+        """
+        return json.loads(
+            super().json(
+                include=include,
+                exclude=exclude,
+                by_alias=by_alias,
+                exclude_unset=exclude_unset,
+                exclude_defaults=exclude_defaults,
+                exclude_none=exclude_none,
+                encoder=encoder,
+                models_as_dict=models_as_dict,
+                **dumps_kwargs,
+            )
+        )
+
+    @public_api
+    def dict(
```

Comment:
I also don't think we should need to touch `dict`. This should be handled in `json` or maybe also `yaml`, but I think using `dict` means it can be dropped from the model internally.

Reply:
@NathanFarmer As per above, I initially agree and will attempt it in the next pull request (it is a good idea to try, even if it does not work!).  Thanks.